src/main/java/org/elasticsearch/index/query/TermQuery.java

```diff
@@ -0,0 +1,205 @@
+     *  - code wise two classes are merged into one which is nice
+     *  - filter and query can maybe share once class and we add a `toFilter(QueryParserContenxt ctx)` method that returns a filter and by default return a `new QueryWrapperFilter(toQuery(context));`
+     */
+
+    public static final String NAME = "term";
+
+    private String fieldName;
+
+    private Object value;
+
+    private float boost = 1.0f;
+
+    private String queryName;
+
+    public TermQuery() {
+    }
+
+    public TermQuery(String fieldName, Object value) {
+        this.fieldName = fieldName;
+        this.value = value;
+    }
+
+    public TermQuery(String fieldName, Object value, String queryName) {
+        this.fieldName = fieldName;
+        this.value = value;
+        this.queryName = queryName;
+    }
+
+    public String getFieldName() {
+        return fieldName;
+    }
+
+    public Object getValue() {
+        return value;
+    }
+
+    public float getBoost() {
+        return boost;
+    }
+
+    public String getQueryName() {
+        return queryName;
+    }
+
+    public void setBoost(float boost) {
+        this.boost = boost;
+    }
+
+    public void setQueryName(String queryName) {
+        this.queryName = queryName;
+    }
+
+    @Override
+    public void readFrom(StreamInput in) throws IOException {
+        fieldName = in.readString();
+        value = in.readGenericValue();
+        boost = in.readFloat();
+        queryName = in.readOptionalString();
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeString(fieldName);
+        out.writeGenericValue(value);
+        out.writeFloat(boost);
+        out.writeOptionalString(queryName);
+    }
+
+    @Override
+    public void doXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject(NAME);
+        if (boost == 1.0f && queryName == null) {
+            builder.field(fieldName, value);
+        } else {
+            builder.startObject(fieldName);
+            builder.field("value", value);
+            if (boost != 1.0f) {
+                builder.field("boost", boost);
+            }
+            if (queryName != null) {
+                builder.field("_name", queryName);
+            }
+            builder.endObject();
+        }
+        builder.endObject();
+    }
+
+    @Override
+    public String[] names() {
+        return new String[]{TermQuery.NAME};
+    }
+
+    @Override
+    public Query parse(QueryParseContext parseContext) throws IOException, QueryParsingException {
+        TermQuery query = new TermQuery();
+        query.fromXContent(parseContext);
+        return query.toQuery(parseContext);
+    }
+
+    public void fromXContent(QueryParseContext context) throws IOException {
```

Comment:
Okay, I think I understand why it is this way.

What I am concerned about is the different ways that a `TermQuery` is constructed here, there's:

```java
new TermQuery(actualField, actualValue)
(new TermQuery()).fromXContent(context)
(new TermQuery()).parse(context) // <-- weird that this is not static
```

What I think would be better is maybe static methods that generate new versions for all except the plain construction version:

```java
new TermQuery(actualField, actualValue)
TermQuery.fromXContent(context) // <-- static, returns new TermQuery
TermQuery.parse(context) // <-- static, returns new TermQuery
```

I dunno, maybe it's a gut feeling :), but the current implementation feels very "loose" and *too* flexible in what the "correct" way to create a new TermQuery, making the methods static instead of mutating the current object feels more functional (in both senses of the word!) to me.

I *personally* would rather have `TermQuery()` constructor be private, but I guess that's an entirely different discussion about builders versus non-builders...